Testing a dependency within CI

I'm trying to set up testing for a dependency (see https://github.com/sublimelsp/lsp_utils/pull/70) within a Github Actions.

There are some problems with that.

By default the CI script bootstraps Package Control with `ignore_vcs_packages` set to `true`. Having that option enabled with a cloned dependency in Packages results in PC showing an error dialog:

![Screenshot 2021-06-19 at 21 43 57](https://user-images.githubusercontent.com/153197/122653910-39a69100-d148-11eb-973a-2baffe8c275e.png)

and UnitTesting script failing the `install_package_control` step with:

```
Skipping git package lsp_utils since the setting "ignore_vcs_packages" is set to true
missing dependencies:
lsp_utils
Timeout: Fail to install Package Control.
Error: Process completed with exit code 1.
```
source: https://github.com/sublimelsp/lsp_utils/runs/2865424911

With `ignore_vcs_packages` set to `false` it also errors out with:

```
Error executing: /usr/bin/git symbolic-ref -q HEAD
Working directory: /github/home/.config/sublime-text/Packages/lsp_utils
VCS-based packages can be ignored by changing the "ignore_vcs_packages" setting to true.
Sublime Text will need to be restarted once the setting is changed.
missing dependencies:
lsp_utils
Timeout: Fail to install Package Control.
Error: Process completed with exit code 1.
```
source: https://github.com/sublimelsp/lsp_utils/runs/2865746622

I'm looking for some ideas on how to handle this.

--- a/sbin/ci.sh
+++ b/sbin/ci.sh
@@ -22,16 +22,15 @@ else
 fi
 
 Bootstrap() {
+    local SkipPackageCopy="$1"
+
     # Disable warnings about detached HEAD
     # https://stackoverflow.com/questions/36794501
     git config --global advice.detachedHead false
 
     # Copy plugin files to Packages/<Package> folder.
-    if [ ! -d "$STP/$PACKAGE" ]; then
-        # symlink does not play well with coverage
-        echo "copy the package to sublime package directory"
-        mkdir -p "$STP/$PACKAGE"
-        cp -r ./ "$STP/$PACKAGE"
+    if [ -z "$SkipPackageCopy" ]; then
+        copyTestedPackage
     fi
 
     local UT_NAME="UnitTesting"
@@ -115,6 +114,15 @@ cloneRepositoryTag() {
     gitGetHeadRevisionName "$DEST"
 }
 
+copyTestedPackage() {
+    if [ ! -d "$STP/$PACKAGE" ]; then
+        # symlink does not play well with coverage
+        echo "copy the package to sublime package directory"
+        mkdir -p "$STP/$PACKAGE"
+        cp -r ./ "$STP/$PACKAGE"
+    fi
+}
+
 InstallPackage() {
     local DEST="$STP/$1"
     local PreferredTag="$2"
@@ -173,6 +181,9 @@ echo "Running command: ${COMMAND} $@"
 case $COMMAND in
     "bootstrap")
         Bootstrap "$@"
+        ;;
+    "copy_tested_package")
+        copyTestedPackage "$@"
         ;;
     "install_package")
         InstallPackage "$@"

--- a/sbin/ci.sh
+++ b/sbin/ci.sh
@@ -30,7 +30,7 @@ Bootstrap() {
 
     # Copy plugin files to Packages/<Package> folder.
     if [ -z "$SkipPackageCopy" ]; then
-        copyTestedPackage
+        CopyTestedPackage
     fi
 
     local UT_NAME="UnitTesting"
@@ -114,7 +114,11 @@ cloneRepositoryTag() {
     gitGetHeadRevisionName "$DEST"
 }
 
-copyTestedPackage() {
+CopyTestedPackage() {
+    local OverwriteExisting="$1"
+    if [ -d "$STP/$PACKAGE" ] && [ -n "$OverwriteExisting" ]; then
+        rm -rf "${STP:?}/${PACKAGE:?}"
+    fi
     if [ ! -d "$STP/$PACKAGE" ]; then
         # symlink does not play well with coverage
         echo "copy the package to sublime package directory"
@@ -183,7 +187,7 @@ case $COMMAND in
         Bootstrap "$@"
         ;;
     "copy_tested_package")
-        copyTestedPackage "$@"
+        CopyTestedPackage "$@"
         ;;
     "install_package")
         InstallPackage "$@"

